spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -36,6 +53,10 @@ def __init__(self, *args, **kwargs):
         self._input_ready = False
         self._last_pdb_cmd = ''
         self._pdb_line_num = 0
+        self.pdb_history_file = PdbHistory()
+        self._control.history = [
```

Comment:
Why does `history` have to be an attribute of `_control`? Why not using an attribute here like `pdb_history`?

Reply:
_control still needs to control the history because it must be able to modify the text. For example, when I press up, the cirrent text is removed, and a new text is replaced.

Reply:
Ok, got it.